pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -744,6 +744,22 @@ func findNextInitContainerToRun(pod *v1.Pod, podStatus *kubecontainer.PodStatus)
 		return nil, nil, true
 	}
 
+	// If any of the main containers have status, then all init containers must
+	// after executed at some point in the past.  However, they could be removed
+	// from the container runtime now, and if we proceed, it would appear as if they
+	// never ran and will re-execute improperly.
```

Comment:
some text correction advice:

```suggestion
	// If any of the main containers have status, then all init containers must
	// have been executed at some point in the past.  However, they could have been removed
	// from the container runtime by now by GC, and if we proceed, it would appear as if they
	// never ran and will re-execute improperly.
```